js/helpers/ng-tracks.js

```
@@ -0,0 +1,185 @@
+angular.module('Clock').service('$tracks', ['$config','$audio', function($config, $audio) {
+    var listeners = [];
+
+    function trigger(event, time) {
+        if(event instanceof Array) {
+            event.forEach(event => trigger(event, time));
+            return;
+        }
+
+        if(typeof event !== 'string') {
+            event = String(event);

```

don't use type constructors. Instead, this should probably throw an error